src/lib/storage/index/group_key/group_key_index.cpp

```diff
@@ -20,9 +20,9 @@ GroupKeyIndex::GroupKeyIndex(const std::vector<std::shared_ptr<const BaseSegment
   Assert((segments_to_index.size() == 1), "GroupKeyIndex only works with a single segment.");
 
   // 1) Initialize the index structures
-  // 1a) Set the index_offset to size of the dictionary + 1 (plus one to mark the ending position)
+  // 1a) Set the index_offset to size of the dictionary + 2 (plus one for null and plus one to mark the ending position)
   //     and set all offsets to 0
-  _index_offsets = std::vector<size_t>(_indexed_segments->unique_values_count() + 1u, 0u);
+  _index_offsets = std::vector<size_t>(_indexed_segments->unique_values_count() + 2u, 0u);
```

Comment:
Add a comment explaining that and how the rest of NULL value handling happens automagically?

Reply:
This has nothing to do with the current PR but now that I see it: the `_index_offsets` points into the `_index_postings`. The `_index_postings` has the same size as the indexed segment which should be at maximum `ChunkOffset` which is a 4 byte instead an 8 byte (`size_t`) integer, right? Could we save memory here?

Reply:
Maybe you could also add the explanation to the schematic explanation in the header file.

Reply:
> Maybe you could also add the explanation to the schematic explanation in the header file.

https://github.com/hyrise/hyrise/commit/068db8c262ffea70e14801ccdbfb89f00f5085b5

> This has nothing to do with the current PR but now that I see it: the `_index_offsets` points into the `_index_postings`. The `_index_postings` has the same size as the indexed segment which should be at maximum `ChunkOffset` which is a 4 byte instead an 8 byte (`size_t`) integer, right? Could we save memory here?

In the worst-case scenario, each value of the segment is a distinct value. Therefore `_index_offsets` has the same size as `_index_postings`. So `ChunkOffset` should be sufficient, shouldn't it?

Reply:
https://github.com/hyrise/hyrise/pull/1772/commits/a05d56d2a1df68c3dddd95fe71b776f5b9cdaf9d